lib/dpl/provider/cloud_foundry.rb

```diff
@@ -23,8 +23,10 @@ def needs_key?
       end
 
       def push_app
-        context.shell "./cf push#{manifest}"
-        context.shell "./cf logout"
+        error 'Failed to push app' unless context.shell("./cf push#{manifest}")
```

Comment:
This `context.shell` call appears to return `false` in specs. Could you look into this?